mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -114,7 +114,7 @@ abstract class StreamingLinearAlgorithm[
     if (model.isEmpty) {
       throw new IllegalArgumentException("Model must be initialized before starting prediction.")
     }
-    data.map(model.get.predict)
```

Comment:
Ah! Thanks Xiangrui, not sure how this slipped in. 

But even with this change, declaring model = None in the superclass is still a problem. With that declaration, and this change, we get a `None.get` exception during any predictions. Curiously, if we replace the `data.mapValues(x => model.get.predict(x))` with the following:
```
data.transform{rdd =>
    val latest = model.get
    rdd.mapValues(x => latest.predict(x))
}
```
It works in a running example (test data reflects updated model), but fails in the new unit tests (that might be a race condition).

With the subclass shadowing, it works in all cases. So I do think we need both.